### PR TITLE
Setup a networkplugin-syncer deployment for OVNKubernetes

### PR DESCRIPTION
**Pull request by mangelajo**
Signed-off-by: Miguel Angel Ajo <majopela@redhat.com>

### CODE REVIEW

File: controllers/submariner/netwokplugin_syncer.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+					Labels: labels,
+				},
+				Spec: corev1.PodSpec{
+					TerminationGracePeriodSeconds: &terminationGracePeriodSeconds,
+					Containers: []corev1.Container{
+						{
+							Name:            "submariner-routeagent",
+							Image:           getImagePath(cr, names.NetworkPluginSyncerImage),
+							ImagePullPolicy: images.GetPullPolicy(cr.Spec.Version),
+							// FIXME: Should be entrypoint script, find/use correct file for routeagent
```
Comment (skitt) [inline controllers/submariner/netwokplugin_syncer.go:56]: Copy-pasta?

File: controllers/submariner/netwokplugin_syncer.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+								{Name: "NODE_NAME", ValueFrom: &corev1.EnvVarSource{
+									FieldRef: &corev1.ObjectFieldSelector{
+										FieldPath: "spec.nodeName",
+									},
+								}},
+							},
+						},
+					},
+					ServiceAccountName: "submariner-networkplugin-syncer",
+					// The route agent engine on all nodes, regardless of existing taints
```
Comment (skitt) [inline controllers/submariner/netwokplugin_syncer.go:75]: Same here?

File: controllers/submariner/netwokplugin_syncer.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+	matchLabels := map[string]string{
+		"app": "submariner-networkplugin-syncer",
+	}
+
+	nReplicas := int32(1)
+	terminationGracePeriodSeconds := int64(1)
+	maxUnavailable := intstr.FromString("100%")
+
+	routeAgentDaemonSet := &appsv1.Deployment{
```
Comment (skitt) [inline controllers/submariner/netwokplugin_syncer.go:30]: This isn’t the route agent, is it?

**Review by skitt — CHANGES_REQUESTED** [review]

File: controllers/submariner/netwokplugin_syncer.go (hunk 3 above)
Comment (mangelajo) [reply controllers/submariner/netwokplugin_syncer.go:30]: ooops

File: controllers/submariner/netwokplugin_syncer.go (hunk 1 above)
Comment (mangelajo) [reply controllers/submariner/netwokplugin_syncer.go:56]: yupppp

**Review by mangelajo — COMMENTED** [review]
thank you @skitt handling those comments.

File: controllers/submariner/netwokplugin_syncer.go (hunk 3 above)
Comment (mangelajo) [reply controllers/submariner/netwokplugin_syncer.go]: Done

File: controllers/submariner/netwokplugin_syncer.go (hunk 1 above)
Comment (mangelajo) [reply controllers/submariner/netwokplugin_syncer.go]: Done

File: controllers/submariner/netwokplugin_syncer.go (hunk 2 above)
Comment (mangelajo) [reply controllers/submariner/netwokplugin_syncer.go]: Done

**Comment by mangelajo** [comment]
fixing the conflicts now..

**Review by tpantelis — APPROVED** [review]

**Review by skitt — APPROVED** [review]

**Pull request merged**